bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/modbus/SolarmanProtocolFactory.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarman.internal.modbus;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.solarman.internal.SolarmanLoggerConfiguration;
+
+/**
+ * @author Peter Kretz - Added RAW Modbus for LAN Stick
+ */
+@NonNullByDefault
```

Comment:
[nit][style] This can be rewritten as:

```
@NonNullByDefault
public class SolarmanProtocolFactory {

    public static SolarmanProtocol createSolarmanProtocol(SolarmanLoggerConfiguration solarmanLoggerConfiguration) {
        return switch (solarmanLoggerConfiguration.getSolarmanLoggerMode()) {
            case RAWMODBUS -> new SolarmanRawProtocol(solarmanLoggerConfiguration);
            case V5MODBUS, null -> new SolarmanV5Protocol(solarmanLoggerConfiguration);
        };
    }
}
```

Reply:
I will change thist.

Reply:
I checked it, an `case V5MODBUS null` is only availabe on source level 21 so the 17 build fails. I changed it to:
`    public static SolarmanProtocol createSolarmanProtocol(SolarmanLoggerConfiguration solarmanLoggerConfiguration) {
        return switch (solarmanLoggerConfiguration.getSolarmanLoggerMode()) {
            case RAWMODBUS -> new SolarmanRawProtocol(solarmanLoggerConfiguration);
			case V5MODBUS -> new SolarmanV5Protocol(solarmanLoggerConfiguration);
			case null -> new SolarmanV5Protocol(solarmanLoggerConfiguration);
        };
    }`